overlord/snapstate/backend/copydata.go

```diff
@@ -86,16 +92,137 @@ func (b Backend) UndoCopySnapData(newInfo, oldInfo *snap.Info, _ progress.Meter,
 }
 
 // ClearTrashedData removes the trash. It returns no errors on the assumption that it is called very late in the game.
-func (b Backend) ClearTrashedData(oldSnap *snap.Info, opts *dirs.SnapDirOptions) {
-	dirs, err := snapDataDirs(oldSnap, opts)
+func (b Backend) ClearTrashedData(oldSnap *snap.Info) {
+	dataDirs, err := snapDataDirs(oldSnap, nil)
 	if err != nil {
 		logger.Noticef("Cannot remove previous data for %q: %v", oldSnap.InstanceName(), err)
 		return
 	}
 
-	for _, d := range dirs {
+	opts := &dirs.SnapDirOptions{HiddenSnapDataDir: true}
+	hiddenDirs, err := snapDataDirs(oldSnap, opts)
+	if err != nil {
+		logger.Noticef("Cannot remove previous data for %q: %v", oldSnap.InstanceName(), err)
+		return
+	}
+
+	// this will have duplicates but the second remove will just be ignored
+	dataDirs = append(dataDirs, hiddenDirs...)
+	for _, d := range dataDirs {
 		if err := clearTrash(d); err != nil {
 			logger.Noticef("Cannot remove %s: %v", d, err)
 		}
 	}
 }
+
+func (b Backend) HideSnapData(snapName string) error {
+	postMigrationOpts := &dirs.SnapDirOptions{HiddenSnapDataDir: true}
+
+	users, err := allUsers(nil)
+	if err != nil {
+		return err
+	}
+
+	for _, usr := range users {
```

Comment:
nitpick: I wouldn't abbreviate "user" to "usr": you don't gain much, and it reminds of `/usr`.

Reply:
I usually do it to avoid conflicts with the "user" package (although it might not be necessary in this case, I didn't check)